controllers/metal3.io/preprovisioningimage_controller.go

```diff
@@ -214,6 +291,20 @@ func setImage(generation int64, status *metal3.PreprovisioningImageStatus, url s
 	return changed
 }
 
+func setUnready(generation int64, status *metal3.PreprovisioningImageStatus, message string) bool {
```

Comment:
nit: is unready actually a word? :)

Reply:
Only [since 1066](https://en.wiktionary.org/wiki/unready), although there was also [this guy](https://en.wikipedia.org/wiki/%C3%86thelred_the_Unready).